scamp/scamp-boot.c

```diff
@@ -26,34 +25,40 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-
 #include "sark.h"
 #include "scamp.h"
 
-
 //------------------------------------------------------------------------------
 
 // This currently sends an image of 28kB at most.
 // Images up to 32kB are possible.
 
+//! The address of the buffer used for booting cores
 #define BOOT_BUF (DTCM_BASE + 0x8000)
 
 // BLOCK_COUNT * BYTE_COUNT must be < 32kB
+//! The number of blocks in the image to boot
 #define BLOCK_COUNT     28      // From 1-256
+//! The number of words in a block (1kB)
 #define WORD_COUNT      256     // From 1-256
+//! The number of bytes in a block
 #define BYTE_COUNT      (WORD_COUNT * sizeof(uint))
 
+//! FF phases, used in nearest neighbour packets.
+enum boot_phases {
+    FF_START_PHASE_1 =          0x01000003,
+    FF_BLOCK_START_PHASE_1 =    0x02000003,
+    FF_BLOCK_DATA_PHASE_1 =     0x03000003,
+    FF_BLOCK_END_PHASE_1 =      0x04000003,
+    FF_CONTROL_PHASE_1 =        0x05000103
+};
 
-#define FF_START_PHASE_1                0x01000003
-#define FF_BLOCK_START_PHASE_1          0x02000003
-#define FF_BLOCK_DATA_PHASE_1           0x03000003
-#define FF_BLOCK_END_PHASE_1            0x04000003
-#define FF_CONTROL_PHASE_1              0x05000103
-
+//! Target monitor
+// TODO: What does this mean?
```

Comment:
Seems to be to do with nearest-neighbour targeting during boot, but if it's not on the datasheet then probably best leaving as a TODO.

Reply:
I _think_ it's about the same protocol as the standard flood fill. (There's an awful lot not stated on the datasheet.)